src/components/OSCALCatalogGroup.test.js

```diff
@@ -1,15 +1,34 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
 import OSCALCatalogGroup from "./OSCALCatalogGroup";
+import { externalCatalogTestData } from "../test-data/ControlsData";
+import { revFourCatalog } from "../test-data/Urls";
 
-test("OSCALCatalog displays control groups", () => {
-  const testGroup = {
-    id: "ac",
-    class: "family",
-    title: "Access Control",
-    controls: [{ id: "ac-1" }],
-  };
-  render(<OSCALCatalogGroup group={testGroup} />);
-  const result = screen.getByText("Access Control");
-  expect(result).toBeVisible();
-});
+function testUrlOSCALCatalog(url) {
+  const externalUrl = url.searchParams.get("url");
+  if (!externalUrl) {
+    test("OSCALCatalog displays control groups", () => {
+      const testGroup = {
+        id: "ac",
+        class: "family",
+        title: "Access Control",
+        controls: [{ id: "ac-1" }],
+      };
+      render(<OSCALCatalogGroup group={testGroup} />);
+      const result = screen.getByText("Access Control");
+      expect(result).toBeVisible();
+    });
+    return;
+  }
+
+  test("ExternalOSCALCatalog displays control groups", () => {
+    render(<OSCALCatalogGroup group={externalCatalogTestData} />);
+    const result = screen.getByText("External Access Control");
+    expect(result).toBeVisible();
+  });
+}
+
+const url = new URL("https://www.oscal-catalog.com");
+testUrlOSCALCatalog(url);
+url.searchParams.set("url", revFourCatalog);
```

Comment:
I'm not sure this is really testing the functionality we're adding.

This looks like it would be the equivalent of someone pasting a URL of `https://www.oscal-catalog.com?url=https://raw.githubusercontent.com/usnistgov/...` into the URL input and click reload, not grabbing it from the browser's `window.location`.

Note that we don't need to test getting the URL from the browser at the component level though.

Reply:
> Might be more readable, maybe not, interested in what others think.

When looking at the code I thought the same thing.

> While those additional tests may be valuable, we might want to consider whether we want to duplicate all tests with different expectations, or restructure the existing tests to follow a pattern of "for this URL, here are the expected values for the different elements", keeping in mind that some sources may not be expected to render certain elements at all.

I also wonder if it is a good idea to test for specific things to be found when loading external components that, as we have seen before, change pretty frequently.

Reply:
So does it make sense to even add more tests with this issue?

Reply:
I think we want to make sure that we:
1. _do_ test documented and public functionality of the application
2. _do not_ test non-deterministic behavior of the application

I wonder if we should `export` `getRequestedUrl` and find a way to mock `window` or `window.location` (or set their attributes or whatever) so that we can at least make sure that we are parsing/retrieving it correctly. Maybe this test is pretty trivial but it's a good thing that could be a source of regressions in the future. Or someone may introduce a feature that causes the app to crash when `url` isn't present.

I'll defer to someone with stronger QA and testing knowledge than me. Perhaps @rgauss or @snorouzzadeh can provide some suggestions on the specifics here.

Reply:
Having played around with our tests a bit, I think currently we have 3 tests that rely on fetching data outside of our mock test data prior to this PR. I think we should really try not to need external data when running tests as they are already pretty slow as is. Whether or not we load our own test data or data from an external source really should not make the page have any significant differences we would want to test for. 

I agree that it would make a lot more sense to check `getRequestedUrl` works for the loader and allow our other tests to stay the same.

Reply:
For testing, I planned on creating a separate folder for tests related to non-component code and creating tests in that folder. That would allow us to separate component tests with tests on other functionality such as grabbing the url from a browser.

Reply:
Could the test for grabbing a `url` just be in it's own `OSCALLoader.test.js` file? I wonder if we need a ton of tests to see the functionality of using a `url` parameter. 

I am curious what @rgauss and @kylelaker think about this. Adding a ton of testing functionality for a minor change seems unneeded here. This has definitely brought up some questions on how we could test external files, but I feel like deciding how to do that and if we should might be out of scope.

Reply:
I'm in agreement on putting it in its own file, that's what I have at the moment. However, I didn't create an `OSCALLoader.test.js` file because it seemed like from the original comment that we should not test this in a component test file.

Reply:
And there are not many tests to write, as we just need to test if `getRequestedUrl` works with no url parameter and when there is a url parameter.